corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js

```diff
@@ -14,14 +14,15 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", function () {
         initialPageData = hqImport("hqwebapp/js/initial_page_data");
 
     // special format handled by CaseSearch API
-    var encodeValue = function (model, value, searchForBlank) {
+    var encodeValue = function (model, searchForBlank) {
+            var value = model.get('value');
```

Comment:
The bug was less significant/visible than it could have been, because `encodeValue` was using the HTML input's value instead of the model's value. This meant that most of the time, things worked out okay, because the HTML value and model value are usually in sync. One time they're not is when there's an error, because that causes the input to get re-rendered, and if the model's value isn't set, re-rendering will blank it out.

The source of truth for the value ought to be the model itself, so I updated this code to pull from the model instead of from HTML. This is the riskier part of this PR.

Reply:
It does bother me a lot how we mix jquery/DOM stuff with JS models